tests/packet_encoding_test.c

```diff
@@ -288,22 +288,61 @@ PACKET_TEST(CONNACK, connack, &s_test_connack_init, NULL, NULL)
 /*****************************************************************************/
 /* Publish                                                                   */
 
-static int s_test_publish_init(struct packet_test_fixture *fixture) {
+static int s_test_publish_qos0_dup_init(struct packet_test_fixture *fixture) {
 
     /* Init packet */
     ASSERT_SUCCESS(aws_mqtt_packet_publish_init(
         fixture->in_packet,
-        false,
+        false /* retain */,
+        AWS_MQTT_QOS_AT_MOST_ONCE,
+        true /* dup */,
+        aws_byte_cursor_from_array(s_topic_name, TOPIC_NAME_LEN),
+        0,
+        aws_byte_cursor_from_array(s_payload, PAYLOAD_LEN)));
+
+    /* Init buffer */
+    /* clang-format off */
+    aws_byte_buf_write_u8(
+        &fixture->buffer,
+        (AWS_MQTT_PACKET_PUBLISH << 4) /* Packet type bits 7-4 */
+        | (1 << 3) /* DUP bit 3 */
+        | (AWS_MQTT_QOS_AT_MOST_ONCE << 1) /* QoS bits 2-1 */
+        | (0 << 0) /* RETAIN bit 0 */);
```

Comment:
Shouldn't this be 1 << 0?

Reply:
retain is set to false in this test, so it's correct to be zero here.

Reply:
it's correct, but yeah, bonkers looking. I'll remove the useless shift